pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -269,18 +273,64 @@ func (is *InstanceService) InstanceCreate(name string, config *openstackconfigv1
 		}
 	}
 	userData := base64.StdEncoding.EncodeToString([]byte(cmd))
-	createOpts = servers.CreateOpts{
+	clusterTag := ""
+	for _, tag := range config.Tags {
+		if strings.Contains(tag.Name, "ClusterID") {
+			clusterTag = tag.Name + tag.Value
+			break
+		}
+	}
+
+	var ports_list []servers.Network
+	for _, net := range nets {
```

Comment:
So, I should first check if it exists and then replace the tags?

Reply:
It is not just about tags. It is about all resources. Before creating a new port, you should check for an existing one. I know… this looks pretty ugly... but the code can fail for whatever reason at any point in time. Even if only the node it runs on dies in the exact moment when a port has been created, but the machine not. The most un-nice thing IMHO is that we must use names to find potential existing resources.
Another example for failing is... parts of OpenStack are not available. Let's assume the Neutron Network API does work, but the Nova Compute API does not work at the time.
This will result in
```
Call to InstanceCreate:
Create port
Create VM => return err

## Reconciler kicks in.
Call to InstanceCreate:
Create port
Create VM => return err

## Reconciler kicks in.
...
```
So the code would create a port at each reconciliation loop. Workflow must be something like

````
Call to InstanceCreate:
Use existing port OR Create new port
Create VM => return err

Call to InstanceCreate
Use existing port
Create VM => return err
```

Reply:
yes yes! I got this! preparing the new version checking for ports on the given net with the name